dist/function.py

```diff
@@ -189,8 +189,139 @@ def function_declaration():
             tfile.close()
             compare_srcfile(tmp_file, name)
 
+# A function definition.
+class FunctionDefinition:
+    def __init__(self, name, module, source):
+        self.name = name
+        self.module = module
+        self.source = source
+        self.used_outside_of_file = False
+        self.used_outside_of_module = False
+
+# Complain if a "wti" function is used outside of its intended scope, or if a "wt" function is used
+# only within its module.
+def function_scoping():
```

Comment:
It took a lot of attention to follow exactly what this is doing. It looks great. I would appreciate a bunch more comments throughout this function.

Reply:
Sounds good; I added more comments in [7f37108](https://github.com/wiredtiger/wiredtiger/pull/10667/commits/7f37108e0ab8c074783c4ceae7b4c7efcbe41532). Hopefully they are indeed helpful!